clippy_lints/src/methods/join_absolute_paths.rs

```diff
@@ -0,0 +1,30 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::is_type_diagnostic_item;
+use rustc_ast::ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::symbol::sym::Path;
+
+use super::JOIN_ABSOLUTE_PATHS;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, join_arg: &'tcx Expr<'tcx>) {
+    let ty = cx.typeck_results().expr_ty(expr).peel_refs();
+    if is_type_diagnostic_item(cx, ty, Path)
+        && let ExprKind::Lit(spanned) = &join_arg.kind
+        && let LitKind::Str(symbol, _) = spanned.node
+        && (symbol.as_str().starts_with('/') || symbol.as_str().starts_with('\\'))
```

Comment:
Does `\\` do this on unix as well? Or is it normalized to `/` immediately? If it doesn't, we could say this behavior would only be present on windows to remove any potential confusion.

Reply:
On unix it just accepts it as a file name

```rs
    let path = Path::new("C:\\Users");
    let path2 = path.join("\\Ducks");
    println!("{path2:#?}"); -> "C:\\Users/\\Ducks"
```

This behavior actually makes sense to me. I'll add a note to the lint description. I wouldn't add it to the lint emission, as it's already quite verbose.

Reply:
Can we use `[has_root](https://github.com/rust-lang/rust-clippy/pull/11453#method.has_root)` to check ?

Reply:
@lengyijun I assume you mean [`Path::has_root()`](https://doc.rust-lang.org/std/path/struct.Path.html#method.has_root)?

I don't think that it would be good to be platform specific here. I can imagine that several projects just run Clippy on linux without specifying `--all-targets`. This assumption comes from the fact that I often forget to specify that flag in the CI. There might also be users, who only run Clippy locally. However, this can lead to false positives, even if I don't believe that anybody wants to start their folder name with `\\` on linux. So, I think we should keep the current behavior, but move the lint to `suspicious`, to have the lint warn by default and not deny. Does that sound good to everyone?